python/tidy.py

```diff
@@ -342,23 +334,36 @@ def check_rust(file_name, contents):
         if match and not (line.startswith("use") or line.startswith("pub use")):
             yield (idx + 1, "missing space after {")
 
+        # check extern crates
+        if line.startswith("extern crate "):
+            crate_name = line[len("extern crate "):-1]
+            indent = len(original_line) - len(line)
+            if indent not in prev_crate:
+                prev_crate[indent] = ""
+            if prev_crate[indent] > crate_name:
+                yield(idx + 1, decl_message.format("extern crate declaration")
+                      + decl_expected.format(prev_crate[indent])
+                      + decl_found.format(crate_name))
+            prev_crate[indent] = crate_name
```

Comment:
No, it's very differerent from `use`. There is no limit on the order, on the contrary of `use` imports. So the sorting is for the whole file.